desk/app/chat.hoon

```diff
@@ -1668,6 +1685,45 @@
           (flatten q.story)
       ==
     ~
+  ++  ca-transfer-channel
+    |=  [new-group=flag:g new=flag:c tim=time]
+    =/  old=log:c  log.chat
+    =/  writ-log
+      =-  +:-
+      %^  (dip:log-on:c @)  log.chat  ~
+      |=  [st=@ =time =diff:c]
+      :_  [%.n st]
+      ::  only keep writs
+      ?.  ?=(%writs -.diff)  ~
+      `diff
+    =/  filtered-log
+        =-  +:-
+        %^  (dip:log-on:c @)  log.chat  ~
+        |=  [st=@ =time =diff:c]
+        :_  [%.n st]
+        ::  keep non-writ events
+        ?.  ?=(%writs -.diff)  `diff
+        ::  only keep writs after time
+        ?.  (gth time tim)  ~
+        `diff
+    =+  .^(=group:g %gx (weld ca-groups-scry /noun))
+    =/  =channel:g  (~(got by channels.group) [%chat flag])
```

Comment:
```suggestion
    =+  .^(=group:g %gx (weld ca-groups-scry /noun))
    =/  =channel:g  (~(got by channels.group) [%chat flag])
    ?>  !=(group.channel new-group)
```

Should probably assert we're actually moving it to a different group (because otherwise this archive, too, is just auto-joined).

Likewise for making sure we're moving it to a non-existent channel, not the same exact channel, etc.